Add cli command to list datasets

BenGalewsky opened the pull request:
This PR depends on the [Service PR 907](https://github.com/ssl-hep/ServiceX/pull/907)

Add a new command to the CLI

```bash
servicex datasets list -b testing1
```

Also accepts an optional argument `--did-finder` to filter on a specific DID Finder such as `rucio` or `user`.

## Example
```bash
% servicex datasets list -b testing2 --did-finder rucio                       
                                                                               ServiceX Datasets                                                                               
┏━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━┓
┃ ID ┃ Name                                                                                                                ┃ Files ┃ Size    ┃ Status   ┃ Created             ┃
┡━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━┩
│ 1  │ rucio://user.mtost:user.mtost.singletop.p6026.Jun13                                                                 │ 27    │ 4,833MB │ complete │ 2024-09-09T22:27:30 │
│ 15 │ rucio://user.mtost:user.mtost.700349.Sh.DAOD_PHYS.e8351_s3681_r13144_r13146_p6026.Jul13_less_jet_and_new_GN?files=7 │ 7     │ 1,360MB │ complete │ 2024-11-05T21:11:23 │
└────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴───────┴─────────┴──────────┴─────────────────────┘
```

ponyisi reviewed (commented):
Could you add tests for the missing coverage in the non-`app/` files? (`servicex_adapter.py`, `servicex_client.py`)

BenGalewsky commented:
@ponyisi - I added a new file for testing `servicex_client` and added a test for the get_datasets method
I also added a test for the case where get_datasets in the `servicex_adapter` throws an auth error

ponyisi commented on servicex/servicex_adapter.py:
```diff
@@ -109,6 +109,22 @@ def get_code_generators(self):
+                           url=f"{self.url}/servicex/datasets",
+                           params=params)
+
+            if r.status_code == 403:
```
Should add code to handle other status codes in a future PR
